homeassistant/components/arve/entity.py

```diff
@@ -0,0 +1,39 @@
+"""Arve base entity."""
+
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ArveCoordinator
+
+
+class ArveDeviceEntity(CoordinatorEntity[ArveCoordinator]):
+    """Defines a base Arve device entity."""
+
+    _attr_has_entity_name = True
+    _attr_available = True
```

Comment:
Should we declare the available property and set it to only be available when `super().available` (which means the coordinator call succeeded) and when `serial_number` is present in `coordinator.data`?